### PR TITLE
Use Python generator by default, use Ruby as fallback

### PR DESCRIPTION
**Pull request by seirl**

### CODE REVIEW

File: tools/generator/rules.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def cxx_args(value) -> str:
+    return ", ".join("{} {}".format(cxx_type(type_), name)
+                     for [name, type_, _] in value)
+
+
+def camel_case(value: str) -> str:
+    """Convert a snake case identifier to a upper camel case one"""
+    return "".join(i.capitalize() for i in value.split("_"))
```
Comment (zopieux) [inline tools/generator/rules.py:20]: Nit: move outside the cxx_ stuff.

File: tools/generator/rules.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                "interface.cc", "rules.hh", "rules.cc", "entry.cc",
+                "game_state.cc", "game_state.hh"]:
+        env.get_template(gen + ".jinja2").stream(game=game).dump(os.path.join(
+            out_dir, gen))
+
+    for action in game["function"]:
+        if "fct_action" in action and action["fct_action"]:
+            (env.get_template("action_template.hh.jinja2")
+             .stream(action=action)
+             .dump(os.path.join(out_dir, "action_")
```
Comment (zopieux) [inline tools/generator/rules.py:54]: Please refactor this by splitting the dir and filename in dedicated variables. It's not very readable. Same below.

**Comment by Shaac** [comment]
We need to add `python-jinja` as dependency.
Both in the doc, and in the PKGBUILD.

**Review by Shaac — CHANGES_REQUESTED** [review]
We need to add `python-jinja` as a dependency.
Both in the doc, and in the PKGBUILD.

**Pull request merged**